### ISSUE TITLE
f_get_from_runs super slow with sets

### ISSUE THREAD

**Issue by maharjun** [SmokinCaterpillar/pypet#42]
From what I see, `f_get_from_runs` function is super slow if the results are stored in result sets. This is because, it appears to check for a particular run name in every run set, and in the case the set doesn't contain the run, then the slow, tree traversal search is performed instead of dict search which leads to infeasably slow variable retrieval times,

I think this can be fixed in one of the following ways:

1. By calculating the set number for each run (since it is always 'run_set_{run_no_by_1000}' and only checking these run sets
2. Get the run_name node using max_depth=1 (or 0? not sure how the depth works) and then extract variables from this.

**Comment by SmokinCaterpillar** [comment]
Have you tried using `max_depth` in your search? Does that speed up the search?

I guess there is a simple speed up like this for the code base: 

```
        try:

            if len(self._run_parent_groups) > 0:
                for run_name in self.f_iter_runs():
                    # Iterate over all runs
                    value = None
                    already_found = False
                    for run_parent_group in self._run_parent_groups.values():
                         if run_name not in run_parent_group._children:
                            continue
                         ...
```
I'll try this, today.

**Comment by SmokinCaterpillar** [comment]
Btw, can you manually add this ```if run_name not in run_parent_group._children: continue``` to your pypet version (I think in line 887 of trajectory.py) and check if it solves your problem? I'll include the fix then in the next version.

**Comment by SmokinCaterpillar** [comment]
has been fixed in 0.4.1
